flink-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/bucketing/BucketingSink.java

```diff
@@ -388,7 +389,7 @@ public void initializeState(FunctionInitializationContext context) throws Except
 		}
 
 		OperatorStateStore stateStore = context.getOperatorStateStore();
-		restoredBucketStates = stateStore.getSerializableListState("bucket-states");
+		this.restoredBucketStates = stateStore.getListState(new ListStateDescriptor("bucket-states", State.class));
```

Comment:
This will not actually work because `getListState(..., State.class)` should be created a `GenericTypeInfo`, which is incompatible with the `JavaSerializer` used before internally here.